### PR TITLE
Feat/improve calibration curve

### LINKED ISSUE #488: Calibration curve display produces flooding log
**Describe the bug**
When displaying the Calibration curve, the log is flooded with the message
"Making the calibrators data for plotting"

**To Reproduce**
1. Open Example "HPLC_UV_Standards"
2. Run Workflow
3. Click on View> Calibrators
4. On Transition explorer select "Antranilicacid", the curve is displayed
5. The log is flooding with the message "Making the calibrators data for plotting"

**Expected behavior**
The message should appear only when showing the plot the first time, or when changing one parameter that involves some refresh of the plot.

**logs**
[CurveFlooding.txt](https://github.com/AutoFlowResearch/SmartPeak/files/7852632/CurveFlooding.txt)

**Version information**
1.18, windows

### LINKED ISSUE #498: Improving Calibrators plot
**Describe the bug**
The calibrator Plot needs some improvments.

**To Reproduce**
1. Open the HPLC_UV_Standards example
2. Click on Action > run workflow
3. Click on "View" > Calibrators
4. Select 5-HTP in the transitions explorer

image

**Expected behavior**
- Not all the points (5HTP-pts, the blue points in the screenshot) are displayed. only the ones that have not been excluded by the calibration curve algorithm are displayed. we would like to have all the points displayed. excluded points and non excluded points may be displayed with a different shape.
- Having the point behind the line is not convenient. On another side, having them on the top can lead to confusion. maybe they can be the same color, but with a different shape.
- Add the possibility to hide the points, and/or the curve.
- Add some margin on top and right
- Add a zoomer item as other plots do
- Add some details about the curve: the parameters used and the one deducted: transformation_model, transformation_model_param_y_weight, ...

**Version information**
1.18.

### PR DESCRIPTION
**Pull request by bertrandboudaud**
Improving the Calibration Plot:

- All points are displayed: the ones that fit the curve (circle) and the ones that don't fit the curve "outer points" (shape : cross)
- Possibility to hide the curve/points/outer points
- Add some margin on top and right
- Use the zooming feature of the Calibration plot
- Add some details about the curve: the parameters used and the one deducted: transformation_model, transformation_model_param_y_weight, ...

CalibrationCurve

### CODE REVIEW

File: src/smartpeak/include/SmartPeak/core/SequenceSegmentHandler.h [hunk 1]
```diff
@@ -112,7 +112,17 @@ namespace SmartPeak
     const std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>>&
     getComponentsToConcentrations() const;
 
-private:
+    void setOuterComponentsToConcentrations(
```
Comment (dmccloskey) [inline src/smartpeak/include/SmartPeak/core/SequenceSegmentHandler.h]: I recommend changing "Outer" to "Outlier" for all of the method and variable names.

File: src/smartpeak/include/SmartPeak/core/SequenceSegmentHandler.h [hunk 2]
```diff
@@ -126,5 +136,6 @@ namespace SmartPeak
     std::shared_ptr<OpenMS::MRMFeatureQC> feature_rsd_estimations_ = nullptr;  ///< Percent RSD estimations; shared between all raw data handlers in the sequence segment
     std::shared_ptr<OpenMS::MRMFeatureQC> feature_background_estimations_ = nullptr;  ///< Background interference estimations; shared between all raw data handlers in the sequence segment
     std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>> components_to_concentrations_;
+    std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>> outer_components_to_concentrations_;
```
Comment (dmccloskey) [inline src/smartpeak/include/SmartPeak/core/SequenceSegmentHandler.h]: I recommend changing `outer_components_to_concentration` to `outlier_components_to_concentration`

File: src/smartpeak/include/SmartPeak/core/SessionHandler.h [hunk 3]
```diff
@@ -336,14 +336,36 @@ namespace SmartPeak
...
+
+    /*
+    @brief Calibration data structure, result of call to setCalibratorsScatterLinePlot
+    */
+    struct CalibrationData
+    {
+      std::vector<std::vector<float>> conc_raw_data;
+      std::vector<std::vector<float>> feature_raw_data;
+      std::vector<std::vector<float>> outer_conc_raw_data;
+      std::vector<std::vector<float>> outer_feature_raw_data;
```
Comment (dmccloskey) [inline src/smartpeak/include/SmartPeak/core/SessionHandler.h]: I recommend changing "outer" to "outlier"

File: src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp [hunk 4]
```diff
@@ -124,13 +128,36 @@ namespace SmartPeak
         LOGW << "Warning: '" << row.getComponentName() << "' cannot be analysed.\n";
         continue;
       }
-      // find the optimal calibration curve for each component
 
+      // Compute outer points
```
Comment (dmccloskey) [inline src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp:132]: Looks good.

File: src/widgets/source/ui/CalibratorsPlotWidget.cpp [hunk 5]
```diff
@@ -26,29 +26,225 @@
...
+        for (int i = 0; i < calibration_data_.conc_raw_data.size(); ++i) {
+          assert(calibration_data_.conc_raw_data.at(i).size() == calibration_data_.feature_raw_data.at(i).size());
+          ImPlot::PushStyleVar(ImPlotStyleVar_Marker, ImPlotMarker_Circle);
+          ImPlot::PlotScatter((calibration_data_.series_names.at(i)).c_str(),
+                               calibration_data_.conc_raw_data.at(i).data(),
+                               calibration_data_.feature_raw_data.at(i).data(),
+                               calibration_data_.conc_raw_data.at(i).size());
+        }
+      }
+      if (show_outer_points_)
```
Comment (dmccloskey) [inline src/widgets/source/ui/CalibratorsPlotWidget.cpp]: I recommend changing this to `show_outlier_points_`

**Review by dmccloskey — APPROVED** [review]
Hi Bertrand.  I went through this a little quick, but the logic and new methods added to compute the outlier points look good.  If you don't mind using "outlier" points instead of "outer" points I think that will make a bit more sense to analytical chemists.  I noted where I saw "outer" in the code.  Please feel free to merge once the naming is taken care of.

**Comment by bertrandboudaud** [comment]
> Hi Bertrand. I went through this a little quick, but the logic and new methods added to compute the outlier points look good. If you don't mind using "outlier" points instead of "outer" points I think that will make a bit more sense to analytical chemists. I noted where I saw "outer" in the code. Please feel free to merge once the naming is taken care of.

ok, i'll do it. thanks.

**Pull request merged**